sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/NexmarkLauncher.java

```diff
@@ -1294,76 +1295,75 @@ private boolean isSql() {
   }
 
   private NexmarkQueryModel getNexmarkQueryModel() {
-    List<NexmarkQueryModel> models = createQueryModels();
+    Map<NexmarkQueryName, NexmarkQueryModel> models = createQueryModels();
     return models.get(configuration.query);
   }
 
   private NexmarkQuery getNexmarkQuery() {
-    List<NexmarkQuery> queries = createQueries();
-
-    if (configuration.query >= queries.size()) {
-      return null;
-    }
-
+    Map<NexmarkQueryName, NexmarkQuery> queries = createQueries();
     return queries.get(configuration.query);
   }
 
-  private List<NexmarkQueryModel> createQueryModels() {
+  private Map<NexmarkQueryName, NexmarkQueryModel> createQueryModels() {
     return isSql() ? createSqlQueryModels() : createJavaQueryModels();
   }
 
-  private List<NexmarkQueryModel> createSqlQueryModels() {
-    return Arrays.asList(null, null, null, null, null, null, null, null, null, null, null, null);
+  private Map<NexmarkQueryName, NexmarkQueryModel> createSqlQueryModels() {
+    return ImmutableMap.of();
   }
 
-  private List<NexmarkQueryModel> createJavaQueryModels() {
-    return Arrays.asList(
-        new Query0Model(configuration),
-        new Query1Model(configuration),
-        new Query2Model(configuration),
-        new Query3Model(configuration),
-        new Query4Model(configuration),
-        new Query5Model(configuration),
-        new Query6Model(configuration),
-        new Query7Model(configuration),
-        new Query8Model(configuration),
-        new Query9Model(configuration),
-        null,
-        null,
-        null);
+  private Map<NexmarkQueryName, NexmarkQueryModel> createJavaQueryModels() {
+    return ImmutableMap.<NexmarkQueryName, NexmarkQueryModel>builder()
+        .put(NexmarkQueryName.ZERO, new Query0Model(configuration))
+        .put(NexmarkQueryName.ONE, new Query1Model(configuration))
+        .put(NexmarkQueryName.TWO, new Query2Model(configuration))
+        .put(NexmarkQueryName.THREE, new Query3Model(configuration))
+        .put(NexmarkQueryName.FOUR, new Query4Model(configuration))
+        .put(NexmarkQueryName.FIVE, new Query5Model(configuration))
+        .put(NexmarkQueryName.SIX, new Query6Model(configuration))
+        .put(NexmarkQueryName.SEVEN, new Query7Model(configuration))
+        .put(NexmarkQueryName.EIGHT, new Query8Model(configuration))
+        .put(NexmarkQueryName.NINE, new Query9Model(configuration))
+        .build();
```

Comment:
A lot better than the list indeed ! Using a list with null values seemed weak and error prone.